### PR TITLE
OpenVINO EP dGPU updates

### PR DESCRIPTION
**Pull request by mayavijx**
Updated the runtime device type option to accept any GPU names like, GPU_FP16, GPU_FP32 , GPU.1_FP16, GPU.1_FP32, GPU.0_FP16, GPU.0_FP32

### CODE REVIEW

**Comment by sfatimar** [comment]
Please resolve conflicts

File: onnxruntime/core/providers/openvino/backends/backend_factory.cc [hunk 1]
```diff
@@ -18,7 +18,7 @@ BackendFactory::MakeBackend(const ONNX_NAMESPACE::ModelProto& model_proto,
   std::string type = global_context.device_type;
   if (type.find("HDDL") != std::string::npos) {
     return std::make_shared<VADMBackend>(model_proto, global_context, subgraph_context);
-  } else if (type == "CPU" || type == "GPU" || type == "MYRIAD" || type.find("HETERO") != std::string::npos ||
+  } else if (type == "CPU" || type.find("GPU") != std::string::npos || type == "MYRIAD" || type.find("HETERO") != std::string::npos ||
```
Comment (sfatimar) [inline onnxruntime/core/providers/openvino/backends/backend_factory.cc:21]: Please resolve all cpplint issues

Comment (mayavijx) [reply onnxruntime/core/providers/openvino/backends/backend_factory.cc]: Resolved cpplint issues.

**Comment by preetha-intel** [comment]
LGTM. Merging the branch.

**Pull request merged**